### PR TITLE
feat: add async-await tag

### PR DESCRIPTION
**Pull request by vcokltfre**
Async await tag as approved in #1585 

image

### CODE REVIEW

File: bot/resources/tags/async-await.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+To call an async function we can either `await` it, or run it in an event loop which we get from `asyncio`.
+
+To create a coroutine that can be used with asyncio we need to define a function using the async keyword:
+```py
+async def main():
+    await something_awaitable()
+```
+Which means we can call `await something_awaitable()` directly from within the function. If this were a non-async function this would have raised an exception like: `SyntaxError: 'await' outside async function`
+
+To run the top level async function from outside of the even loop we can get an event loop from `asyncio`, and then use that loop to run the function:
```
Comment (ToxicKidz) [inline bot/resources/tags/async-await.md:16]:
```suggestion
To run the top level async function from outside of the event loop we can get an event loop from `asyncio`, and then use that loop to run the function:
```

File: bot/resources/tags/async-await.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+```py
+from asyncio import get_event_loop
+
+async def main():
+    await something_awaitable()
+
+loop = get_event_loop()
+loop.run_until_complete(main())
+```
+Note that in the `run_until_complete()` where we appear to be calling `main()`, this does not execute the code in `main`, rather it returns a `Coroutine` object which is then handled and run by the event loop via `run_until_complete()`.
```
Comment (ToxicKidz) [inline bot/resources/tags/async-await.md:26]:
```suggestion
Note that in the `run_until_complete()` where we appear to be calling `main()`, this does not execute the code in `main`, rather it returns a `coroutine` object which is then handled and run by the event loop via `run_until_complete()`.
```

File: bot/resources/tags/async-await.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+To learn more about asyncio and it's use, see the [asyncio documentation](https://docs.python.org/3/library/asyncio.html).
```
Comment (ToxicKidz) [inline bot/resources/tags/async-await.md:28]:
```suggestion
To learn more about asyncio and its use, see the [asyncio documentation](https://docs.python.org/3/library/asyncio.html).
```

**Review by ToxicKidz — CHANGES_REQUESTED** [review]
Hey, this is a good tag, just a couple of suggestions. Is it worth mentioning the use of `asyncio.run` as well?

**Review by Xithrius — APPROVED** [review]
Yes.

**Review by ToxicKidz — APPROVED** [review]
:shipit:

**Pull request merged**